ANTLR/BaseVisitor.py

```diff
@@ -192,11 +198,73 @@ def visitReverse(self, ctx: GoPoParser.ReverseContext):
         return self.visitChildren(ctx)
 
     def visitSort(self, ctx: GoPoParser.SortContext):
-        is_descending = False if self.visit(ctx.getChild(2)) == '+' else True
+        is_descending = False if ctx.op.type == GoPoParser.PLUS else True
         self.tmp_memory['list'].sort(reverse=is_descending)
 
         return self.visitChildren(ctx)
 
+    def visitMapOpWithNum(self, ctx:GoPoParser.MapOpWithNumContext):
+        value = self.convert_str_to_numeric(self.visit(ctx.getChild(3)))
+        function = lambda_two_args_by_operator[ctx.op.type]
+        self.tmp_memory['list'] = [function(x, value) for x in self.tmp_memory['list']]
+
+        return self.visitChildren(ctx)
+
+    def visitMapOpWithoutNum(self, ctx:GoPoParser.MapOpWithoutNumContext):
+        function = lambda_one_arg_by_operator[ctx.op.type]
+        self.tmp_memory['list'] = list(map(function, self.tmp_memory['list']))
+
+        return self.visitChildren(ctx)
+
+    def visitDropLast(self, ctx:GoPoParser.DropLastContext):
+        self.returned_value_from_list_function = self.tmp_memory['list'].pop()
+
+        return self.visitChildren(ctx)
+
+    def visitDropWithIndex(self, ctx:GoPoParser.DropWithIndexContext):
+        index = self.convert_str_to_numeric(self.visit(ctx.getChild(2)))
+        self.returned_value_from_list_function = self.tmp_memory['list'].pop(index)
+
+        return self.visitChildren(ctx)
+
+    def visitCount(self, ctx: GoPoParser.CountContext):
+        self.returned_value_from_list_function = len(self.tmp_memory['list'])
+
+        return self.visitChildren(ctx)
+
+    def visitSum(self, ctx:GoPoParser.SumContext):
+        self.returned_value_from_list_function = sum(self.tmp_memory['list'])
+
+        return self.visitChildren(ctx)
+
+    def visitContains(self, ctx:GoPoParser.ContainsContext):
+        value = self.convert_str_to_numeric(self.visit(ctx.getChild(2)))
+        self.returned_value_from_list_function = True if value in self.tmp_memory['list'] else False
+
+        return self.visitChildren(ctx)
+
+    def visitIs_empty(self, ctx: GoPoParser.Is_emptyContext):
+        self.returned_value_from_list_function = True if len(self.tmp_memory['list']) == 0 else False
+
+        return self.visitChildren(ctx)
+
+    def visitClear(self, ctx: GoPoParser.ClearContext):
+        self.tmp_memory['list'].clear()
+
+    def visitPrint(self, ctx: GoPoParser.PrintContext):
+        print(self.visit(ctx.getChild(2)))
+
+    def visitForeach(self, ctx:GoPoParser.ForeachContext):
+        temp_variable = ctx.getChild(2).symbol.text
+        for value in self.tmp_memory['list']:
+            self.memory[temp_variable] = value
+            type = ctx.getChild(4).start.type
```

Comment:
I really don't know for what purpose it is done?

`self.visit(ctx.getChild(4))` should be enough

Reply:
We have to specify which function to use (`ctx.stat() or ctx.stat_block_newline()`)

Reply:
yes, but getChild(4) is stat or stat_block_newline

just see how it is defined in .g4
`foreach	    : FOREACH OPAR ID '->' (stat | stat_block_newline)  CPAR;`

Reply:
ok, I changed it